docs/reference/certificates.md

```diff
@@ -0,0 +1,131 @@
+---
+layout: docwithnav
+title: Kubernetes PKI certificates and requirements
+permalink: /certs/
+approvers:
+- efrost
+---
+
+# Introduction
+
+A tool like [kubeadm][kubeadm] can be used to generate your certificates for you.
+But there are also reasons to generate these certificates yourself.
+This allows you to keep your private keys off of the api server, for enhanced security.
+
+# Certificates in Kubernetes
+
+Kuberentes uses TLS extensively for authentication.
+
+## Where they're used
+
+Kubernetes uses PKI for the following operations:
+1. Client certificates for the kubelet to authenticate to the API server
+2. Server certificate for the API server endpoint
+3. Client certificates for administrators of the cluster authenticating to the API server
+4. Client certificates for the API server to talk to the kubelets
+5. Client certificate for the API server to talk to etcd
+6. Client certificate/kubeconfig for the controller manager to talk to the API server
+7. Client certificate/kubeconfig for the scheduler to talk to the API server.
+8. Client and server certificates for the [front-proxy][proxy]
+
+
+In addition, etcd uses mutual TLS for authenticating clients and peers (in an HA cluster).
+
+## Where to find them
+
+Kubernetes stores certificates by default in `/etc/kubernetes/pki`. All paths in this documentation are relative to that directory.
+
+## Certificate configuration
+
+There are multiple levels of control for kubernetes certificates. 
+At the basic level, a tool like [kubeadm][kubeadm] can generate all required certificates.
+If that isn't desireable, a single root CA, controlled by an adminstrator, can create multiple _intermediate CAs_, and delegate all further creation to Kubernetes itself. 
+
+
+The CAs needed for this to work are:
+
+| path                   | Default CN                | description                      |
+|------------------------|---------------------------|----------------------------------|
+| ca.crt,key             | kubernetes-ca             | the kubernetes general CA        |
+| etcd/ca.crt,key        | kubernetes-front-proxy-ca | For all ETCD-related functions   |
+| front-proxy-ca.crt,key | etcd-ca                   | for the [front-end proxy][proxy] |
+
+If you don't wish to copy these private keys to your API servers, you can generate all certificates yourself. 
+In that case, you will also need to make all of these:
+
+
+
+[2]: For a liveness probe, if self-hosted
+
+| Default CN                    | Parent CA                 | O (in Subject) | kind   | hosts (SAN)                                 |
+|-------------------------------|---------------------------|----------------|--------|---------------------------------------------|
+| kube-etcd                     | etcd-ca                   |                | server | localhost, 127.0.0.1                        |
```

Comment:
Since v1.11 now uses etcd 3.2.x, we currently have to create the etcd serving certificate with both server and client usages. Upstream etcd issue tracking this is here: https://github.com/coreos/etcd/issues/9785